### PR TITLE
SOC-5552: Add the property type to the space entity.

### PR DESCRIPTION
**Pull request by ahamdi**
Added the property type to the space entity.
This property was present with JCR implementation, in fact it is needed for custom implementations of  space creation when we need to have many space types.

### CODE REVIEW

**Comment by boubaker** [comment]
ok for me

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/SpaceStorageTest.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ protected Space getSpaceInstance(int number) {
     space.setDisplayName("my space test " + number);
     space.setPrettyName(space.getDisplayName());
     space.setRegistration(Space.OPEN);
+    space.setType("testSpaceType");
```
Comment (boubaker) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/SpaceStorageTest.java]: space.setType is set twice here.

Comment (ahamdi) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/SpaceStorageTest.java]: Done

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/SpaceStorageTest.java [hunk 2]
```diff
@@ -1524,6 +1524,9 @@ public void testGetSpaceByPrettyName() throws Exception {
     assertEquals("space.getDescription() must return: " + space.getDescription(),
                  space.getDescription(),
                  foundSpaceList.getDescription());
+    assertEquals("space.getType() must return: " + space.getType(),
```
Comment (boubaker) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/SpaceStorageTest.java]: IMO you should set a custom type to test if it's stored as required:
    Space space = this.getSpaceInstance(number);
    space.setType("customeType");

**Comment by ahamdi** [comment]
Done

**Comment by ahamdi** [comment]
Merged

**Pull request closed without merging**